core/tests/test_utils.py

```diff
@@ -824,9 +822,14 @@ def put_json(self, url, payload, csrf_token=None, expected_status_int=200):
         self.assertEqual(json_response.status_int, expected_status_int)
         return self._parse_json_response(json_response, expect_errors)
 
-    def get_csrf_token_from_response(self, response):
-        """Retrieve the CSRF token from a GET response."""
-        return re.search(CSRF_REGEX, response.body).group(1)
+    def get_csrf_token(self):
```

Comment:
Could we also call this get_new_csrf_token() or generate_new_csrf_token()? I think it's important to be very clear about when we are getting a new token and when we are retrieving an existing one.